airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_report_streams.py

```diff
@@ -554,3 +557,42 @@ def test_read_incremental_with_records_start_date(config):
         records = list(read_incremental(stream, state))
         assert state == {"1": {"reportDate": "20210104"}}
         assert {r["reportDate"] for r in records} == {"20210103", "20210104", "20210105", "20210106"}
+
+
+@pytest.mark.parametrize("state_filter", [["enabled", "archived", "paused"], ["enabled"], None])
```

Comment:
Now, these three tests are the same except for the class name. Use `itertools.product` to generate test cases for all state filter and stream class pairs.

```python
from itertools import product

@pytest.mark.parametrize("state_filter", "klass", product(
    [["enabled", "archived", "paused"], ["enabled"], None],
    [SponsoredDisplayCampaigns, SponsoredProductCampaigns, SponsoredBrandsCampaigns]
))
def test_campaign_state_filer(mocker, config, state_filter, klass):
    ...
    mocker.patch.object(klass, "state_filter", new_callable=mocker.PropertyMock, return_value=state_filter)
    ...
```